packages/core/src/lib/markdown-it/index.js

```diff
@@ -140,13 +142,32 @@ markdownIt.renderer.rules.fence = (tokens, idx, options, env, slf) => {
   // wrap all lines with <span> so we can number them
   str = lines.map((line, index) => {
     const currentLineNumber = index + 1;
```

Comment:
The code block processing is getting quite long here. Should we move them to `Highlighter.js` / `HighlightRule.js`?

I would prefer the former personally as the latter file is also quite long, but rename the file to `highlightRules.js` since its about processing multiple highlight rules. But up to you.

Specifically, maybe `parseHighlightRules` ([L93](https://github.com/MarkBind/markbind/pull/2059/files#diff-1a21f01839ec5b717dd80d08be9066b103f4b387565102d4226766291f8fda9eR93)) and `applyHighlightRules` (here).

Reply:
If going with the latter, would also transfer the functions you have in `Highlighter.js` (inline or move to HighlightRule.js/helper.js). Having a class of static functions seems more like idiomatic Java.